093.mediawiki

```diff
@@ -0,0 +1,326 @@
+
+Graphene maintains this size advantage as block size grows. For example, for a block of ''n=10,000'' transactions, listing 8-bytes of each transaction ID would be 80,000 bytes. With a mempool of ''m=30,000'' transactions, Graphene's cost is 14,482 bytes (or 31,091 bytes when including ordering information).
+
+
+===New Messages===
+Graphene introduces four new messages to the network protocol: '''get_grblk''', '''grblk''', '''get_grblktx''', and '''grblktx'''.
+
+====get_grblk====
+The '''get_grblk''' message (suggested implementation name: '''GET_GRAPHENE''') transmits a single serialized '''CMempoolInfo''' data structure. It is used to both signal the desire to receive a Graphene block and to communicate the number of transactions in the receiver's mempool. 
+
+====grblk====
+The '''grblk''' message (suggested implementation name: '''GRAPHENEBLOCK''') transmits a single serialized '''CGrapheneBlock''' data structure. In the absence of failure or after successful recovery of items from set ''M'' (see above), the message contains sufficient information for the receiver to reconstruct the full block.
+
+====get_grblktx====
+The '''get_grblktx''' (suggested implementation name: '''GET_GRAPHENETX''') message transmits a single serialized '''CRequestGrapheneBlockTx'''. This message is sent in the event that the IBLT subtraction operation succeeded and revealed a non-empty set of missing transaction IDs ''M''.
+
+====grblktx====
+The '''grblktx''' (suggested implementation name: '''GRAPHENETX''') message transmits a single serialized '''CGrapheneBlockTx'''. This message is sent in response to '''get_grblktx'''; it contains full transactions corresponding to any cheap hashes in '''CRequestGrapheneBlockTx''' that also appear in the block.
+
+
+===New data structures===
+Graphene introduces several new data structures to the network protocol: '''CMemPoolInfo''', '''CGrapheneBlock''', '''CIblt''' '''CGrapheneBlockTx''', and '''CRequestGrapheneBlockTx'''.
+
+All data structures use the standard Bitcoin serialization format; some structures additionally use a compact size encoding, which we note in the tables below. Except for '''CGrapheneBlock''', the data structures are comprised of relatively simple c++ constructs, which we detail below. '''CGrapheneBlock''' contains '''CGrapheneSet''', which itself contains Bloom filter and IBLT data structures, denoted '''CBloomFilter''' and '''CIblt''', respectively.
+We describe each of these complex structures separately.
+
+====CGrapheneBlock====
+''CGrapheneBlock'' is the fundamental data structure for Graphene block propagation. Assuming no unrecoverable failures arise, and combined with any missing transactions recovered from the sender, this data structure contains all information necessary to reconstruct the full block. It is sent as part of the '''grblk''' message.
+
+{|
+ |'''Field Name'''||'''Type'''||'''Size'''||'''Purpose'''
+ |-
+ |header||Block header||80 bytes||The header of the block being provided
```

Comment:
", encoded as per the header portion of the BLOCK message"

Reply:
Updated.